keras_nlp/models/bert.py

```diff
@@ -0,0 +1,287 @@
+    }
+    output = encoder(input_data)
+    ```
+    """
+
+    # TODO(bischof): add tests
+
+    def __init__(
+        self,
+        vocab_size,
+        num_layers,
+        hidden_size,
+        num_heads,
+        intermediate_dim,
+        dropout=0.1,
+        max_sequence_length=512,
+        num_segments=2,
+        **kwargs,
+    ):
+
+        initializer_fn = keras.initializers.TruncatedNormal(stddev=0.02)
+
+        # Functional version of model
+        token_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="input_ids"
+        )
+        segment_id_input = keras.Input(
+            shape=(None,), dtype="int32", name="segment_ids"
+        )
+        input_mask = keras.Input(
+            shape=(None,), dtype="int32", name="input_mask"
+        )
+
+        # Embed tokens, positions, and segment ids.
+        token_embedding_layer = keras.layers.Embedding(
+            input_dim=vocab_size,
+            output_dim=hidden_size,
+            name="token_embedding",
+        )
+        token_embedding = token_embedding_layer(token_id_input)
+        position_embedding = PositionEmbedding(
+            initializer=initializer_fn,
+            sequence_length=max_sequence_length,
+            name="position_embedding",
+        )(token_embedding)
+        segment_embedding = keras.layers.Embedding(
+            input_dim=num_segments,
+            output_dim=hidden_size,
+            name="segment_embedding",
+        )(segment_id_input)
+
+        # Sum, normailze and apply dropout to embeddings.
+        x = keras.layers.Add(
+            name="embeddings/sum",
+        )((token_embedding, position_embedding, segment_embedding))
+        x = keras.layers.LayerNormalization(
+            name="embeddings/layer_norm",
+            axis=-1,
+            epsilon=1e-12,
+            dtype=tf.float32,
+        )(x)
+        x = keras.layers.Dropout(
+            dropout,
+            name="embeddings/dropout",
+        )(x)
+
+        # Apply successive transformer encoder blocks.
+        for i in range(num_layers):
+            x = TransformerEncoder(
+                num_heads=num_heads,
+                intermediate_dim=intermediate_dim,
+                activation=lambda x: keras.activations.gelu(
+                    x, approximate=True
+                ),
+                dropout=dropout,
+                kernel_initializer=initializer_fn,
+                name="transformer/layer_%d" % i,
+            )(x, padding_mask=input_mask)
+
+        # Construct the two BERT outputs. The pooled output is a dense layer on
+        # top of the [CLS] token.
+        sequence_output = x
+        pooled_output = keras.layers.Dense(
+            hidden_size,
+            activation="tanh",
+            name="pooled_dense",
+        )(x[:, CLS_INDEX, :])
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs={
+                "input_ids": token_id_input,
```

Comment:
We need a way to support use case that only input tokens are provided. 

To be specific, users should be able to do this:
```
model = keras_nlp.models.Bert(...)
tokenized_input = tokenizer("This is a test sentence")
model_output = model(tokenized_input)
```

Reply:
Discussed offline that segment ids are always necessary for Bert arch. We will provide preprocessing utils in followup PRs.